electronics/scripts/export_schematic.py

```diff
@@ -72,13 +79,31 @@ def eeschema_plot_schematic(output_directory):
     logger.info('Plot')
     xdotool(['key', 'Return'])
 
-    logger.info('Wait before shutdown')
-    time.sleep(2)
+def process_has_file_open(process, file_path):
+    for file in process.open_files():
+        if file.path == file_path:
+           return True
+    return False
+
+def wait_for_plot(eeschema_proc, plot_file, timeout=2):
+    begin = time.time()
+    ps = psutil.Process(eeschema_proc.pid)
+    while not process_has_file_open(ps, plot_file):
```

Comment:
One thing I'm thinking of is what happens if kicad finishes quickly (so the opposite of what we initially saw)? Presumably if kicad was really fast it could open *and* close the file before we even call `wait_for_plot`, in which case it would time out thinking the file was never opened, right?

Not sure how realistic that concern is, but I'm wondering if it wouldn't be more robust to instead just have a single wait loop in this method, which will wait as long as *either* eeschema has the file open OR the output file is missing/0-bytes. What do you think of an approach like that? Then it would initially be looping waiting because the file doesn't exist, and then loop because kicad had the file open, and eventually exit when the file is closed and non-empty. Not sure if it's worth it so I'm curious what you think.